proto/stat_var.proto

```diff
@@ -173,13 +171,20 @@ message GetStatVarPathResponse {
   repeated string path = 1;
 }
 
-message SearchResultSVG {
-  // DCID of the stat var group
+
+message SearchResultEntry {
+  // DCID of the result
   string dcid = 1;
-  // Name of the stat var group
+  // Name of the result
   string name = 2;
+  // The strings in the name that match the search tokens
+  repeated string matches = 3;
```

Comment:
instead of embedding this with each result, can we achieve the same (with a much smaller response) by returning a set of matching words across all results at the SearchStatVarResponse level? For most searches, this set would stay fairly small, linear with the size of the search string (not very long) rather than the search results (usually very long).

In both cases, we would still need to do string matching to highlight the matched terms so the client would be handling the information the same way. wdyt?

Reply:
Agree with this.

Also the search index now is getting bigger too with each token have multiple linked text. I have seen issues with the search index blowing up the mixer memory. Do you think there is also a way to make the record here smaller?

Reply:
Actually, yes, that makes sense. My original thinking was that because of synonyms, not all results will have the same set of matching words. But now that I think about it, since the client side is doing string matching, the client side can figure out which of the words in the set actually match each result.